src/types/enrichment/datasource.ts

```diff
@@ -0,0 +1,43 @@
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as t from "io-ts";
+
+const DBParams = t.type({
+  connectionString: NonEmptyString,
+  query: NonEmptyString,
+  whereConditionId: NonEmptyString,
+});
+
+const APIParams = t.type({
+  authentication: t.partial({
+    apiKeyHeaderName: NonEmptyString,
+    apiKeyHeaderValue: NonEmptyString,
+  }),
+  endpointUrl: NonEmptyString,
+  idParam: NonEmptyString,
+});
+
+const TableStorageParams = t.type({
+  connectionString: NonEmptyString,
+  partitionKey: NonEmptyString,
+  rowKey: NonEmptyString,
+  tableName: NonEmptyString,
+});
+
+const BlobStorageParams = t.type({
+  blobFilename: NonEmptyString,
+  connectionString: NonEmptyString,
+  containerName: NonEmptyString,
+  extension: NonEmptyString,
+});
+
+export const EnrichmentDataSource = t.type({
+  params: t.union([BlobStorageParams, TableStorageParams, APIParams, DBParams]),
+  type: t.union([
+    t.literal("BlobStorage"),
+    t.literal("TableStorage"),
+    t.literal("API"),
+    t.literal("DB"),
+  ]),
+});
```

Comment:
this type doesn' t guarantee type disjoint based on `type` attribute. We should define different types like this:
```
export const BlobStorageEnrichmentDataSource = t.type({
  params: BlobStorageParams,
  type: t.literal("BlobStorage"),
});
```
and then `EnrichmentDataSource` should be defined as union of all Enrichment specific datasource's types